doc/pitfalls/data_leakage.rst

```diff
@@ -0,0 +1,152 @@
+.. _data_leakage:
+
+============
+Data leakage
+============
+
+Data leakage occurs when information that would not be available at prediction
+time, is used when building the model. This results in overly optimsitic
+performance estimates, for example from :ref:`cross validation
+<cross_validation>`, but poorer performance when the model is used
+on actually novel data, for example during production.
+
+A common cause is not keeping the test and train data subsets separate. Test
+data should never be used to make choices about the model but this may
+accidently occur during :ref:`preprocessing` or :ref:`grid_search`.
+
+Although both train and test data subsets should undergo the same preprocessing
+transformation, it is important that stateful transformations only use the
+training subset to determine the 'state'. For example, if you have a
+normalization step whereby you divide by the average value, the average should
+be the average of the train subset, **not** the average of all the data. If the
+test subset was included in the average calculation, information from the test
+subset is influencing the model. Other types of preprocessing such as
+:ref:`impute` and :ref:`polynomial_features` should also only utilize train
+data.
```

Comment:
I would add a note that this anti-pattern is to call fit_transform(X_test) and not transform(X_test).

Reply:
or link to specific examples just below so a clear example is a click away

Reply:
I would go further and not that the anti-pattern is to call `fit(X_test)` and thus this applies to `fit_transform()` as well ;)